### ISSUE TITLE
Use modern OpenGL

### ISSUE THREAD

**Issue by v1993** [cubosphere/cubosphere-code#26]
**Is your feature request related to a problem? Please describe.**
If we want to make it possible to port cubosphere to mobile world, using OpenGL 3.1+ is simply a requiment.

This will be huge work, and lua shaders will be just gone, but it is just required.

**Describe the solution you'd like**
Replace Fixed Pipeline with Programmable one, i.e. GL shaders + VBO.

**Describe alternatives you've considered**
There are two of them:

1. Throw out cubosphere and write cubosphere 2.0 from `int main`. Looking at all arhitecture flaws and problems, it is probably a right way…
2. Just ignore entire problem and use OpenGL 2.0.

**Additional context**
Nope.

**Comment by qwertychouskie** [comment]
Actually, for best compatibility for older Android devices, GLES 2 support is important.

**Comment by cddepppp256** [comment]
Personally, I'd go with rewriting the whole thing. It would be more rewarding than putting it off until OpenGL 2 is basically dead (or later).

The old PC in my bedroom won't like it either way.

**Comment by v1993** [comment]
@cddepppp256 Yeah, I completely agree with you. To achive better compatiblity with mobile devices (which have very limited shader support) and overall portability across modern devices I seriously think about using vulkan. It also will be very useful for me personally to learn it.

**Comment by qwertychouskie** [comment]
The number of Android devices that support Vulkan is quite small (https://www.quora.com/What-kind-of-Android-devices-are-compatible-with-Vulkan).  Honestly for supporting the most devices OpenGL ES 2.0 is necessary at least as a fall-back renderer.  OpenGL ES 3.0 will also run on a decent chunk of devices, but just because a device supports the API doesn't mean it is powerful enough to run a full-fat rendering engine.

In short, the quickest way to get Cubosphere running on the most devices is to tweak the current renderer to also support OpenGL ES 2.0.

P.S. I just had a crazy idea: Cubosphere using WebAssembly and WebGL, running in a browser :)

**Comment by v1993** [comment]
@qwertychouskie Tweaking for OpenGL 3.0 is very, very hard to do. You can try, but overall work is so tedious that I'm not going to do it at all. I don't think that modifying such overall bad designed project that much is good idea, honestly.

As for Vulkan, there are libs which allow using it with OpenGL as backend.
